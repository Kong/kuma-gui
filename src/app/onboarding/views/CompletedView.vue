<template>
  <RouteView>
    <RouteTitle
      :title="t('onboarding.routes.completed.title')"
    />
    <AppView>
      <OnboardingPage>
        <template #header>
          <OnboardingHeading>
            <template #title>
              Go to the dashboard
            </template>
          </OnboardingHeading>
        </template>

        <template #content>
          <div class="gui-preview-image">
            <img src="@/assets/images/kuma_gui.png">
          </div>
        </template>

        <template #navigation>
          <OnboardingNavigation
            next-step="home"
            next-step-title="Complete"
            last-step
            :show-skip="false"
          />
        </template>
      </OnboardingPage>
    </AppView>
  </RouteView>
</template>

<script lang="ts" setup>
import OnboardingHeading from '../components/OnboardingHeading.vue'
import OnboardingNavigation from '../components/OnboardingNavigation.vue'
import OnboardingPage from '../components/OnboardingPage.vue'
import AppView from '@/app/application/components/app-view/AppView.vue'
import RouteTitle from '@/app/application/components/route-view/RouteTitle.vue'
import RouteView from '@/app/application/components/route-view/RouteView.vue'
import { useI18n } from '@/utilities'
const { t } = useI18n()
</script>

<style lang="scss" scoped>
.gui-preview-image {
  display: flex;
  justify-content: center;
}
</style>
