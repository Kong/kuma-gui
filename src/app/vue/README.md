---
section: Packages
---
# vue

Automatic `vue` initialization using our projects dependency injection layer.

