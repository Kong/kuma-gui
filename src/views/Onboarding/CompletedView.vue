<template>
  <OnboardingPage>
    <template #header>
      <OnboardingHeading title="Go to the dashboard" />
    </template>

    <template #content>
      <div class="flex justify-center">
        <img src="@/assets/images/kuma_gui.png">
      </div>
    </template>

    <template #navigation>
      <OnboardingNavigation
        next-step="global-overview"
        next-step-title="Completed"
        last-step
        :show-skip="false"
      />
    </template>
  </OnboardingPage>
</template>

<script lang="ts" setup>
import OnboardingNavigation from '@/views/Onboarding/components/OnboardingNavigation.vue'
import OnboardingHeading from '@/views/Onboarding/components/OnboardingHeading.vue'
import OnboardingPage from '@/views/Onboarding/components/OnboardingPage.vue'
</script>
