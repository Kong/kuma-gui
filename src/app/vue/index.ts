import { createApp } from 'vue'
import {
  createRouter,
  createWebHistory,
} from 'vue-router'

import type { ServiceDefinition } from '@/services/utils'
import { token, createInjections } from '@/services/utils'
import type { Component } from 'vue'
import type { Router, RouteRecordRaw, NavigationGuard } from 'vue-router'

type Token = ReturnType<typeof token>

type VueApp = ReturnType<typeof createApp>
export type PluginDefinition = Parameters<VueApp['use']>
export type ComponentDefinition = [string, Component]

const $ = {
  app: token<(App: Component) => Promise<VueApp>>('vue.app'),
  router: token<Router>('vue.router'),

  components: token('vue.components'),
  plugins: token<PluginDefinition[]>('vue.plugins'),
  routes: token<RouteRecordRaw[]>('vue.routes'),
  routesLabel: token<RouteRecordRaw[]>('vue.routes.label'),
  navigationGuards: token<NavigationGuard[]>('vue.routes.navigation.guards'),
}
export const services = (app: Record<string, Token>): ServiceDefinition[] => {
  return [
    [$.app, {
      service: (
        components: ComponentDefinition[],
        plugins: PluginDefinition[],
      ) => {
        return async (App: Component) => {
          const app = createApp(App)

          plugins.forEach(([...args]) => {
            app.use(...args)
          })

          components.forEach(([name, component]: [string, Component]) => {
            app.component(name, component)
          })

          return app
        }
      },
      arguments: [
        $.components,
        $.plugins,
      ],
    }],
    [$.router, {
      service: (
        env: (str: 'KUMA_BASE_PATH') => string,
        routes: RouteRecordRaw[],
        guards: NavigationGuard[],
      ) => {
        const router = createRouter({
          history: createWebHistory(env('KUMA_BASE_PATH')),
          routes,
        })

        guards.forEach((item) => {
          if (typeof item === 'function') {
            router.beforeEach(item)
          }
        })

        return router
      },
      arguments: [
        app.env,
        $.routes,
        $.navigationGuards,
      ],
    }],

    [token('vue.plugins'), {
      service: (router) => {
        return [
          [router],
        ]
      },
      arguments: [
        $.router,
      ],
      labels: [
        $.plugins,
      ],
    }],

    // TODO(jc): Label decoration. Make it so we can decorate labels
    // Temporarily turn the routesLabel label into routes for decoration
    // purposes
    [$.routes, {
      service: (routes: RouteRecordRaw[]) => routes,
      arguments: [
        $.routesLabel,
      ],
    }],
    //

    // TODO(jc): Make it so we don't need to provide a [] empty default for
    // labels
    [token('application.routes.navigation.guards'), {
      service: () => {
        return []
      },
      labels: [
        $.navigationGuards,
      ],
    }],
  ]
}

export const TOKENS = $
export const [
  useRouter,
] = createInjections(
  TOKENS.router,
)
