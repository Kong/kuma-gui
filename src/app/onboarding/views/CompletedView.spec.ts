import { render } from '@testing-library/vue'

import CompletedView from './CompletedView.vue'

describe('CompletedView.vue', () => {
  it('renders snapshot', () => {
    const { container } = render(CompletedView)

    expect(container).toMatchSnapshot()
  })
})
