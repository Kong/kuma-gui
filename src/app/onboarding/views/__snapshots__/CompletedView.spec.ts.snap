// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedView.vue renders snapshot 1`] = `
<div
  class="route-view"
>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="route-view-title sr-only"
  >
    Navigated to 
  </div>
  
  <!--v-if-->
  <main
    class="app-main-content"
  >
    
    <!--v-if-->
    <section
      class=""
    >
      <!--v-if-->
      
      <div>
        <div
          class="onboarding-container"
        >
          <div
            class="onboarding-container__header"
          >
            
            <div
              class="onboarding-heading"
            >
              <h1
                class="onboarding-title"
              >
                
                 Go to the dashboard 
                
              </h1>
              <!--v-if-->
            </div>
            
          </div>
          <div
            class="onboarding-container__content"
          >
            <div
              class="onboarding-container__inner-content"
            >
              
              <div
                class="gui-preview-image"
              >
                <img />
              </div>
              
            </div>
          </div>
          <div
            class="mt-4"
          >
            
            <div
              class="onboarding-actions"
            >
              <!--v-if-->
              <div
                class="button-list"
              >
                <!--v-if-->
                <a
                  class="k-button medium rounded creation"
                  data-testid="onboarding-next-button"
                  href="/gui/"
                  type="button"
                >
                  
                  <!---->
                  
                  
                  Complete
                  
                  <!---->
                </a>
              </div>
            </div>
            
          </div>
        </div>
        <div
          class="background-image"
        />
      </div>
      
    </section>
    
  </main>
  
</div>
`;
